examples/posix_sockets/udp.c

```diff
@@ -43,6 +50,19 @@ static char server_buffer[SERVER_BUFFER_SIZE];
 static char server_stack[THREAD_STACKSIZE_DEFAULT];
 static msg_t server_msg_queue[SERVER_MSG_QUEUE_SIZE];
 
+/* find first valid network interface */
+#ifdef SOCK_HAS_IPV6
+static uint32_t _get_first_netif(void) {
+#ifdef MODULE_GNRC_SOCK_UDP
+    gnrc_netif_t *iface = gnrc_netif_iter(NULL);
```

Comment:
As `posix_sockets` is not GNRC specific, I would prefer the usage of [`net/netif.h`](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/netif.h).

Reply:
> As this application is by its nature supposed to be independent from the network stack below, we should really carefully think about how we get the network interface:

Thank you very much, I will have a look.

Reply:
If you want to collect bonus points, a POSIX wrapper around `net/netif.h` would be the ideal solution ;-), but for now I would be happy with just using `net/netif.h`

Reply:
I think I can do that, too. [`net/if.h`](https://pubs.opengroup.org/onlinepubs/009619199/netifh.htm#tagcjh_04_02) would be it and looks not too complicated. :)